### PR TITLE
Add basic logs enabled check

### LINKED ISSUE #92104: Azure Monitor Basic Logs: Existing Basic Log queries still run after Basic Logs option is disabled in datasource
### What happened?

After enabling Basic Logs in the Azure Monitor datasource settings, dashboards and panel queries work as expected when panels are set to Basic option. However, after disabling the Basic Logs option in the datasource, dashboards and panels continue to work when the feature has been disabled. This is not the expected behavior. I would expect the datasource setting to be enforced and prevent any basic logs queries from being run.

When the panel is open for editing, the query does not run and this is more in line with the expected behavior.

image

image

### What did you expect to happen?

I expect all existing panels configured with basic logs to not run, or for the queries to fail if datasource setting disables the feature.

### Did this work before?

No.

### How do we reproduce it?

1. Enable Basic Logs in Azure Monitor plugin datasource config
2. Create and save dashboards using Basic Logs queries
3. Set Basic Logs to disabled in Azure Monitor plugin config.
4. Open existing dashboards to see successful queries.

### Is the bug inside a dashboard panel?

_No response_

### Environment (with versions)?

Grafana: 11.1
OS: Windows
Browser: Chrome and Edge

### Grafana platform?

Other

### Datasource(s)?

Azure Monitor

### PR DESCRIPTION
**Pull request by alyssajoyner**
This PR adds a check to ensure that if a user has Basic Logs disabled it will not complete the query. 

Closes #92104

### CODE REVIEW

**Review by bossinc — APPROVED** [review]

File: pkg/tsdb/azuremonitor/loganalytics/utils.go [hunk 1]
```diff
@@ -55,6 +56,10 @@ func meetsBasicLogsCriteria(resources []string, fromAlert bool) (bool, error) {
 		return false, fmt.Errorf("basic Logs queries may only be run against Log Analytics workspaces")
 	}
 
+	if !basicLogsEnabled {
+		return false, fmt.Errorf("basic Logs queries are disabled for this workspace")
```
Comment (aangelisc) [inline pkg/tsdb/azuremonitor/loganalytics/utils.go:60]:
```suggestion
		return false, fmt.Errorf("basic Logs queries are disabled for this data source")
```

**Review by aangelisc — CHANGES_REQUESTED** [review]
Nice work, just need to change which flag we're validating (and perhaps add a supporting test).

File: pkg/tsdb/azuremonitor/loganalytics/utils.go [hunk 2]
```diff
@@ -40,19 +40,24 @@ func AddConfigLinks(frame data.Frame, dl string, title *string) data.Frame {
...
 		return false, fmt.Errorf("basic logs queries cannot be run against multiple resources")
 	}
 
 	if !strings.Contains(strings.ToLower(resources[0]), "microsoft.operationalinsights/workspaces") {
-		return false, fmt.Errorf("basic Logs queries may only be run against Log Analytics workspaces")
+		return false, fmt.Errorf("basic logs queries may only be run against Log Analytics workspaces")
+	}
+
+	if !basicLogsEnabled {
+		return false, fmt.Errorf("basic logs queries are not enabled")
```
Comment (aangelisc) [inline pkg/tsdb/azuremonitor/loganalytics/utils.go:60]:
```suggestion
		return false, fmt.Errorf("basic logs queries are disabled for this data source")
```

File: pkg/tsdb/azuremonitor/loganalytics/azure-log-analytics-datasource_test.go [hunk 3]
```diff
@@ -100,6 +101,7 @@ func TestBuildLogAnalyticsQuery(t *testing.T) {
 	tests := []struct {
 		name                   string
 		fromAlert              bool
+		setBasicLogsEnabled    bool
```
Comment (aangelisc) [inline pkg/tsdb/azuremonitor/loganalytics/azure-log-analytics-datasource_test.go:104]: I think you need to set this value on the `dsInfo` in order to appropriately use it.

File: pkg/tsdb/azuremonitor/loganalytics/utils.go [hunk 4]
```diff
@@ -53,6 +54,10 @@ func meetsBasicLogsCriteria(resources []string, fromAlert bool) (bool, error) {
 	}
 
 	if !strings.Contains(strings.ToLower(resources[0]), "microsoft.operationalinsights/workspaces") {
+		return false, fmt.Errorf("basic logs queries may only be run against Log Analytics workspaces")
```
Comment (aangelisc) [inline pkg/tsdb/azuremonitor/loganalytics/utils.go:57]:
```suggestion
		return false, errorsource.DownstreamError(fmt.Errorf("basic logs queries may only be run against Log Analytics workspaces"))
```

File: pkg/tsdb/azuremonitor/loganalytics/utils.go [hunk 5]
```diff
@@ -53,6 +54,10 @@ func meetsBasicLogsCriteria(resources []string, fromAlert bool) (bool, error) {
...
+	}
+
+	if !basicLogsEnabled {
```
Comment (aangelisc) [inline pkg/tsdb/azuremonitor/loganalytics/utils.go:60]: I'd suggest updating the error on the next line to reflect the actual problem 😊

File: pkg/tsdb/azuremonitor/loganalytics/utils.go [hunk 6]
```diff
@@ -53,7 +54,11 @@ func meetsBasicLogsCriteria(resources []string, fromAlert bool) (bool, error) {
 	}
 
 	if !strings.Contains(strings.ToLower(resources[0]), "microsoft.operationalinsights/workspaces") {
-		return false, errorsource.DownstreamError(fmt.Errorf("basic Logs queries may only be run against Log Analytics workspaces"), false)
+		return false, errorsource.DownstreamError(fmt.Errorf("basic logs queries may only be run against Log Analytics workspaces"))
+	}
+
+	if !basicLogsEnabled {
+		return false, errorsource.DownstreamError(fmt.Errorf("basic Logs queries are disabled for this workspace"), false)
```
Comment (aangelisc) [inline pkg/tsdb/azuremonitor/loganalytics/utils.go:61]:
```suggestion
		return false, errorsource.DownstreamError(fmt.Errorf("basic Logs queries are disabled for this data source"), false)
```

**Review by aangelisc — APPROVED** [review]
LGTM, nice work @alyssabull!

**Pull request merged**